src/utils/Encode.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+// import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+// import "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+
+import "../interfaces/IMultiSend.sol";
+import "../interfaces/ISafe.sol";
+import "../interfaces/IProxyAdmin.sol";
+import "../interfaces/IUpgradeableBeacon.sol";
+
+struct MultisigCall {
+    address to;
+    uint256 value;
+    bytes data;
+}
+
+library Encode {
+    /// Used for state vars:
+    ///
+    /// uint idx;
+    /// mapping(uint => MultisigCall[]) map;
+    bytes32 internal constant IDX_SLOT = keccak256("IDX_SLOT");
+    bytes32 internal constant MAP_SLOT = keccak256("MAP_SLOT");
+
+    /// Used to describe calls from a Gnosis Safe
+    enum Operation {
+        Call,
+        DelegateCall
+    }
+
+    /// Constants for Safe.execTransaction inputs we don't usee
+    uint256 constant SAFE_TX_GAS = 0;
+    uint256 constant BASE_GAS = 0;
+    uint256 constant GAS_PRICE = 0;
+    address constant GAS_TOKEN = address(uint160(0));
+    address payable constant REFUND_RECEIVER = payable(address(uint160(0)));
+
+    /// Dummy types and variables to facilitate syntax, e.g: `Encode.proxyAdmin.upgrade(...)`
+    enum EncProxyAdmin {
+        A
+    }
+    enum EncUpgradeableBeacon {
+        A
+    }
+    enum EncGnosisSafe {
+        A
+    }
+
+    EncProxyAdmin internal constant proxyAdmin = EncProxyAdmin.A;
+    EncUpgradeableBeacon internal constant upgradeableBeacon = EncUpgradeableBeacon.A;
+    EncGnosisSafe internal constant gnosisSafe = EncGnosisSafe.A;
+
+    /// @dev Creates a new, clean `MultisigCall[] storage` pointer, guaranteeing
+    /// any previous pointers will not be overwritten.
+    /// Since we're in a library, we have to use assembly+slot pointers, but the
+    /// high level version of this function equates to:
+    ///
+    /// uint _idx = storage.idx;
+    /// storage.idx++;
+    /// MultisigCall[] storage calls = storage.map[_idx];
+    /// return calls;
+    function newMultisigCalls() internal returns (MultisigCall[] storage) {
+        bytes32 _IDX_SLOT = IDX_SLOT;
+        bytes32 _MAP_SLOT = MAP_SLOT;
+
+        uint256 idx;
+        assembly {
+            idx := sload(_IDX_SLOT)
+            sstore(_IDX_SLOT, add(1, idx))
+        }
+
+        // fn pointer indirection fools the compiler into letting us have
+        // an uninitialized storage pointer
+        function() pure returns (mapping(uint => MultisigCall[]) storage) fn;
```

Comment:
![image](https://github.com/user-attachments/assets/faaf680c-63ed-4626-b30d-9561298acab5)

Reply:
you likey?

Reply:
sol devs would rather do this than just contribute to the compiler LMAO